frameworks/moveos-stdlib/sources/module_store.move

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+/// `module_store` provide object to manage packages and modules.
+module moveos_std::module_store {
+    use std::vector;
+    use std::string::String;
+    use moveos_std::core_addresses;
+    use moveos_std::object::{Self, ObjectID, Object};
+    use moveos_std::tx_context;
+    use moveos_std::signer;
+    use moveos_std::move_module::{Self, MoveModule};
+    use moveos_std::features;
+
+    friend moveos_std::genesis;
+    
+    /// Not allow to publish module
+    const ErrorNotAllowToPublish: u64 = 1;
+
+    /// Allowlist for module function invocation
+    struct Allowlist has key, store {
+        /// Allow list for publishing modules
+        publisher: vector<address>,
+
+    }
+
+    /// Used to store packages.
+    /// A package is an Object, and the package ID is the module address.
+    /// Packages are dynamic fields of ModuleStore.
+    struct ModuleStore has key {
+    }
+
+    /// Used to store modules.
+    /// Modules are the Package's dynamic fields, with the module name as the key.
+    struct Package has key, store {}
```

Comment:
`Package` should not have `store` ability. The owner can not directly transfer the `Package` to others. If we want to support `Package` transfer, we can extend the `transfer_extend` function in `module_store`.